### PR TITLE
PEPPER-1562 . PHI manifest update

### PR DESCRIPTION
**Pull request by ssettipalli**
PEPPER-1562 
PHI manifest update to make collaborator participant Id is included in the scenario where there is tumor only orders.

### CODE REVIEW

**Review by denniscunningham — COMMENTED** [review]
Is there a way to test this via PhiManifestTest?

**Comment by ssettipalli** [comment]
> Is there a way to test this via PhiManifestTest?

tested it locally running DSM and having execution go through new code. 
Considering a very edge case scenario, we should be good. prod has 2 cases that the code execution will go through new code to get it.
Test seem to be focusing on values from elastic and this field is being retrieved from DB

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java [hunk 1]
```diff
@@ -186,6 +190,30 @@ public Boolean isBloodKit(String ddpLabel) {
...
+            SimpleResult dbVals = new SimpleResult("");
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_COLLABORATOR_PARTICIPANT_ID)) {
+                stmt.setString(1, ddpParticipantId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getString(DBConstants.COLLABORATOR_PARTICIPANT_ID);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = new DsmInternalError("Error getting collaborator participant Id ", dbVals.resultException);
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java]: Including the ddpParticipantId in the message will help debugging.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java [hunk 2]
```diff
@@ -186,6 +190,30 @@ public Boolean isBloodKit(String ddpLabel) {
         return booleanCheckFoundAsName(ddpLabel, SQL_IS_BLOOD_KIT_QUERY);
     }
 
+    public String getCollaboratorParticipantId(String ddpParticipantId) {
+        if (StringUtils.isBlank(ddpParticipantId)) {
+            return null;
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java]: Is null a useful return value? I see the caller is immediately assigning this value.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java [hunk 3]
```diff
@@ -186,6 +190,30 @@ public Boolean isBloodKit(String ddpLabel) {
...
+            SimpleResult dbVals = new SimpleResult("");
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_COLLABORATOR_PARTICIPANT_ID)) {
+                stmt.setString(1, ddpParticipantId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getString(DBConstants.COLLABORATOR_PARTICIPANT_ID);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = new DsmInternalError("Error getting collaborator participant Id for participant: " + ddpParticipantId, dbVals.resultException);
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java]: This code effectively swallows this exception.
I suggest either throwing an exception, or logging a ERROR and returning a value the caller can use, perhaps NULL as above.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java [hunk 4]
```diff
@@ -186,6 +190,30 @@ public Boolean isBloodKit(String ddpLabel) {
...
+            return null;
+        }
+
+        SimpleResult results = inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult("");
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_COLLABORATOR_PARTICIPANT_ID)) {
+                stmt.setString(1, ddpParticipantId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getString(DBConstants.COLLABORATOR_PARTICIPANT_ID);
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java]: This code would be clearer if you removed the indirection through `SimpleResults`, and just returned a value here. (You can also add the logging from below here.) That way anything that falls through the bottom of these conditionals can be handled uniformly. 
Likewise you can either catch and rethrow exceptions below, or not handle exceptions at this level, depending on your preferred error handling approach.

**Review by denniscunningham — CHANGES_REQUESTED** [review]

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java (hunk 2 above)
Comment (ssettipalli) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java]: not really needed. cleaned up

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java (hunk 4 above)
Comment (ssettipalli) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDao.java]: refactored and cleaned up

**Review by denniscunningham — APPROVED** [review]
